pylint/checkers/base.py

```diff
@@ -1316,19 +1427,13 @@ def _find_name_group(self, node_type):
         return self._name_group.get(node_type, node_type)
 
     def _raise_name_warning(self, node, node_type, name, confidence):
-        type_label = _NAME_TYPES[node_type][1]
-
-        hint_name = node_type.replace('_', '-') + '-name-hint'
-        hint_part = '%r template' % hint_name
-
-        if self.config.include_naming_hint:
-            hint_rgx = getattr(self.config, hint_name.replace('-', '_'))
-            hint_part += ' (hint: %r)' % hint_rgx
+        type_label = HUMAN_READABLE_TYPES[node_type]
```

Comment:
Have you seen emitted message from current top of this PR? 

For out of the box settings:

    Function name "BADFUNCTION_name" doesn't conform to snake_case naming style_

For user defined settings:

    _Class name "classb" doesn't conform to '(?:(?P<UP>[A-Z]+)|(?P<down>[a-z]+))$' pattern_

I personally like them a lot, and to reintroduce include-naming-hint, I'd effectively replace this verbose and precise definitions with something arbitrary like _doesn't conform to selected naming style_ and _doesn't conform to custom user-defined regular expression_. Personally, I strongly prefer current form.

Include-naming-hint seems like a workaround for older messages where we emitted "Invalid class name 'classb'". With this new approach, where we provide both predefined and advanced mode for users, I don't see much value in keeping it.